macaw-dump/src/Data/Macaw/Dump.hs

```diff
@@ -0,0 +1,35 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE ImportQualifiedPost #-}
+{-# LANGUAGE TypeOperators #-}
+{-# OPTIONS_GHC -Wno-simplifiable-class-constraints #-}
```

Comment:
Well, without suppressing it I see:
```haskell
src/Data/Macaw/Dump.hs:20:3: error: [-Wsimplifiable-class-constraints, Werror=simplifiable-class-constraints]
    • The constraint ‘CCE.IsSyntaxExtension (MS.MacawExt arch)’ matches
        instance macaw-symbolic-0.0.1:Data.Macaw.Symbolic.CrucGen.MacawArchConstraints
                   arch =>
                 CCE.IsSyntaxExtension (MS.MacawExt arch)
          -- Defined in ‘macaw-symbolic-0.0.1:Data.Macaw.Symbolic.CrucGen’
      This makes type inference for inner bindings fragile;
        either use MonoLocalBinds, or simplify it using the instance
    • In the type signature:
        main :: (MS.GenArchInfo MS.LLVMMemory arch,

                 CCE.IsSyntaxExtension (MS.MacawExt arch),
                 MC.ArchConstraints arch,
                 MC.ArchAddrWidth arch ~ EE.RelocationWidth reloc,
                 EE.IsRelocationType reloc) =>
                MAI.ArchitectureInfo arch
                -> MS.GenArchVals mem arch -> MMEP.PLTStubInfo reloc -> IO ()
   |
20 |   ( MS.GenArchInfo MS.LLVMMemory arch
   |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^...
```
But if I remove the `IsSyntaxExtension` constraint, I then get:
```
src/Data/Macaw/Dump.hs:32:32: error: [GHC-39999]
    • Could not deduce ‘Data.Parameterized.TraversableFC.TraversableFC
                          (macaw-symbolic-0.0.1:Data.Macaw.Symbolic.CrucGen.MacawArchStmtExtension
                             arch)’
        arising from a use of ‘MDD.discover’
      from the context: (MS.GenArchInfo MS.LLVMMemory arch,
                         MC.ArchConstraints arch,
                         MC.ArchAddrWidth arch ~ EE.RelocationWidth reloc,
                         EE.IsRelocationType reloc)
        bound by the type signature for:
                   main :: forall arch reloc mem.
                           (MS.GenArchInfo MS.LLVMMemory arch, MC.ArchConstraints arch,
                            MC.ArchAddrWidth arch ~ EE.RelocationWidth reloc,
                            EE.IsRelocationType reloc) =>
                           MAI.ArchitectureInfo arch
                           -> MS.GenArchVals mem arch -> MMEP.PLTStubInfo reloc -> IO ()
        at src/Data/Macaw/Dump.hs:(19,1)-(28,7)
    • In the expression: MDD.discover archInfo archVals cfg
      In a case alternative:
          MDC.CommandDiscover cfg -> MDD.discover archInfo archVals cfg
      In a stmt of a 'do' block:
        case MDC.cliCommand cli of
          MDC.CommandDiscover cfg -> MDD.discover archInfo archVals cfg
          MDC.CommandPlt cfg -> MDP.plt archInfo pltStubInfo cfg
   |
32 |     MDC.CommandDiscover cfg -> MDD.discover archInfo archVals cfg
   |                                ^^^^^^^^^^^^
```
I get a different error if I remove `ArchConstraints`.

Reply:
Huh, I'm surprised that we don't export `MacawArchConstraints`, as it seems like the most direct way to write this would be replacing the `IsSyntaxExtension (MS.MacawExt arch)` constraint with `MacawArchConstraints arch`. But failing that, I'd favor enabling `MonoLocalBinds` rather than disabling the warning wholesale.